### PR TITLE
Scalar Functions Docs Improvements

### PR DESCRIPTION
**Pull request by Alex-Monahan**
This documents many additional scalar functions. Let me know what feedback you have! Some other changes to how functions are documented include:

- Added a Utility Functions page
- Exposed the Catalog Functions page already made by @hawkfish (and made slight tweaks)
- Added a Text Similarity Functions section to the text functions page (open to feedback here!)
- Added alternative syntax to pattern matching docs (Ex: ~~ instead of LIKE)

Remaining TODOs include:

- [ ] Adding the GLOB syntax to the pattern matching docs (upcoming PR in next few days)
- [ ] make_date / make_time functions (I think you had already done some work here @hawkfish?)
- [ ] enum_first / etc. (@pdet - I think you may have some work on those already maybe?)
- [ ] Documenting other types of functions like macros and table functions
- [ ] Automating the construction of the scalar docs. I've built a basic parser for pulling the existing markdown tables that can be used to build a flat file that contains all of the descriptions and examples we currently have. I likely need a bit of help on the exact next steps to take here!
- [ ] Standardizing how we handle aliases. In some cases we have another column for aliases, in others we have another row. Let me know which format you prefer!
- [ ] Splitting up the text functions into sections.

### CODE REVIEW

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks excellent. One minor comment:

File: docs/sql/functions/numeric.md [hunk 1]
```diff
@@ -34,24 +37,34 @@ The table below shows the available mathematical functions.
...
+| `random()` | returns a random number between 0 and 1 | `random()` | various |
 | `round(v numeric, s int)` | round to *s* decimal places | `round(42.4332, 2)` | 42.43 |
 | `setseed(x)` | sets the seed to be used for the random function | `setseed(0.42)` | |
 | `sin(x)` | computes the sin of x | `sin(90)` | 0.8939966636005579 |
 | `sign(x)` | returns the sign of x as -1, 0 or 1 | `sign(-349)` | -1 |
 | `sqrt(x)` | returns the square root of the number | `sqrt(9)` | 3 |
 | `xor(x)` | bitwise XOR | `xor(17, 5)` | 20 |
 | `tan(x)` | computes the tangent of x | `tan(90)` | -1.995200412208242 |
+| `@` | absolute value (parentheses optional if operating on a column) | `@(-2)` | 2 |
+| `!__postfix` | See `!` operator. Factorial of x. Computes the product of the current integer and all integers below it. Function name must be wrapped in double quotes. | `"!__postfix"(5)` | 120 |
```
[the remark this answers is not in the document]
Comment (Alex-Monahan) [reply docs/sql/functions/numeric.md:70]: Sure! It does work if you call that function, but I hope people just use the ! syntax! :-)

**Comment by Alex-Monahan** [comment]
Merged in the other upstream changes and removed !__postfix. Ready to be reviewed again. Thanks!

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**